File: src/main/java/top/chenray/antilitematica/detection/DetectionHandler.java
package top.chenray.antilitematica.detection;

/**
 * Handler for detection events dispatched through {@link DetectionBus}.
 * <p>
 * Implementations should check {@link DetectionBus.DetectionContext#detectionType()}
 * and return {@code true} if they handled the detection (consuming it),
 * or {@code false} to pass it to the next handler in the chain.
 */
@FunctionalInterface
public interface DetectionHandler {

   /**
    * Handle a detection event.
    *
    * @param ctx the detection context
    * @return true if this handler consumed the detection (subsequent handlers are skipped)
    */
   boolean handle(DetectionBus.DetectionContext ctx);
}

File: src/main/kotlin/icu/epochcraft/antilitematica/detection/DetectionHandler.kt
package icu.epochcraft.antilitematica.detection

/**
 * 检测处理器：注册到 [DetectionBus] 的处理链。
 *
 * 按注册顺序依次尝试；返回 true 表示认领本次检测（后续处理器跳过）。
 *
 * @author 阿清
 */
fun interface DetectionHandler {

    /**
     * @param ctx 检测上下文
     * @return true 认领（后续处理器跳过），false 放行给下一个
     */
    fun handle(ctx: DetectionContext): Boolean
}
